packages/@romejs/compiler/lint/rules/react/jsxFragments.ts

```diff
@@ -0,0 +1,38 @@
+import {Path, TransformExitResult} from "@romejs/compiler";
+import {descriptions} from "@romejs/diagnostics";
+import {JSXElement, jsxFragment} from "@romejs/ast";
+import {hasJSXAttribute} from "@romejs/js-ast-utils";
+
+function isStandardFragment(node: JSXElement): boolean {
+	return !!((node.name.type === "JSXReferenceIdentifier" &&
+	node.name.name === "Fragment") ||
+	(node.name.type === "JSXMemberExpression" &&
+	node.name.object.type === "JSXReferenceIdentifier" &&
+	node.name.object.name === "React" &&
+	node.name.property.type === "JSXIdentifier" &&
+	node.name.property.name === "Fragment"));
+}
+
+export default {
+	name: "jsxFragments",
+	enter(path: Path): TransformExitResult {
+		const {node, context} = path;
+
+		if (
+			node.type === "JSXElement" &&
+			isStandardFragment(node) &&
```

Comment:
You can use `doesNodeMatchPattern` here from `js-ast-utils`:

```suggestion
			(doesNodeMatchPattern(node.name, 'Fragment') || doesNodeMatchPattern(node.name, 'React.Fragment')) &&
```

Reply:
```
const Hello = <div><React.Fragment><Foo /></React.Fragment></div>;
const Hello = <React.Fragment><Foo /></React.Fragment>;
function Foo() {
	let bar = <React.Fragment><Foo /></React.Fragment>;
	return bar;
}
function Hello() {
	return <React.Fragment><Foo /></React.Fragment>;
}
```
That affected some of the tests. Would you be okay if I did the following:

```
			node.type === "JSXElement" &&
			(doesNodeMatchPattern(node.name, 'Fragment') || node.name.type === "JSXMemberExpression" && doesNodeMatchPattern(node.name.property, 'Fragment')) &&
			!hasJSXAttribute(node, "key")
```
cleaner than originally but not as clean as your suggestion, but I think that doesNodeMatchPattern doesn't capture React.Fragment from your suggestion.

Reply:
I think this can work if doesNodeMatchPattern handles JSXMemberExpressions and not only JSMemberExpressions. Let me double check

Reply:
Yup! Updating `doesNodeMatchPattern` seems like the right call.